pyiode/objects/scalar.pyx

```diff
@@ -84,15 +85,13 @@ cdef class Scalar:
     # Special methods
 
     def __eq__(self, other: Scalar) -> bool:
-        if not isinstance(other, Scalar):
-            raise TypeError(f"Expected argument of type 'Scalar'.\nGot argument of type '{type(other).__name__}'")
         return self.c_scalar == other.c_scalar
 
     def __str__(self) -> str:
-        return self.c_scalar.to_string().decode()
+        return f"Scalar({_iode_number_to_str(self.value)}, {_iode_number_to_str(self.relax)}, {_iode_number_to_str(self.std)})"
 
     def __repr__(self) -> str:
-        return self.c_scalar.to_string().decode()
+        return str(self)
```

Comment:
I am unsure it works for cython, but for pure Python, the standard is to use the same method for both instead of calling one from the other (avoiding both an extra function call and a method call) by using something like this: 
```python
__repr__ = __str__
```